### PR TITLE
Renamed `databricks_catalog_workspace_binding` to `databricks_workspace_binding`

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Renamed `databricks_catalog_workspace_binding` to `databricks_workspace_binding`

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by nkvuong** [comment]
integration tests passed (failed tests are unrelated)

**Review by edwardfeng-db — APPROVED** [review]

File: catalog/resource_workspace_binding.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			"catalog_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"workspace_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		}}).CoreConfigSchema().ImpliedType()
+}
```
Comment (alexott) [inline catalog/resource_workspace_binding.go:151]: Do we need this here? It should be used only for deprecated resource

File: docs/resources/workspace_binding.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `workspace_id` - ID of the workspace. Change forces creation of a new resource.
+* `securable_name` - Name of securable. Change forces creation of a new resource.
+* `securable_type` - Type of securable. Default to `catalog`. Change forces creation of a new resource.
```
Comment (alexott) [inline docs/resources/workspace_binding.md:38]: I think that we should add possible values for `securable_type`.

File: docs/resources/workspace_binding.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  workspace_id   = databricks_mws_workspaces.other.workspace_id
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `workspace_id` - ID of the workspace. Change forces creation of a new resource.
+* `securable_name` - Name of securable. Change forces creation of a new resource.
```
Comment (alexott) [inline docs/resources/workspace_binding.md:37]: We need to clarify what securable name is - for catalog, and other objects. And maybe add examples besides the catalog

**Review by alexott — COMMENTED** [review]
Overall looks good but we may need to extend docs. Also not sure about migration code for the new resource

**Comment by nkvuong** [comment]
@alexott I've separated the new changes for storage_credential & external_location to a different PR - https://github.com/databricks/terraform-provider-databricks/pull/3678

**Review by alexott — APPROVED** [review]

**Pull request merged**